src/mscorlib/src/System/Environment.cs

```diff
@@ -379,9 +379,9 @@ public static String CurrentDirectory
             [System.Security.SecuritySafeCritical]  // auto-generated
 #endif
             get {
-                StringBuilder sb = new StringBuilder(Path.MAX_PATH);
-                int r = Win32Native.GetSystemDirectory(sb, Path.MAX_PATH);
-                Contract.Assert(r < Path.MAX_PATH, "r < Path.MAX_PATH");
+                StringBuilder sb = new StringBuilder(Path.MaxPath);
```

Comment:
We have not done a holistic sweep to find and fix all places - we have been only fixing what showed on the radar...